src/Aspire.Hosting.Azure/AzureDeployingContext.cs

```diff
@@ -4,17 +4,74 @@
 #pragma warning disable ASPIREAZURE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
 #pragma warning disable ASPIREPUBLISHERS001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
 
+using Aspire.Hosting.Azure.Provisioning;
 using Aspire.Hosting.Azure.Provisioning.Internal;
+using Aspire.Hosting.Publishing;
 
 namespace Aspire.Hosting.Azure;
 
 internal sealed class AzureDeployingContext(
     IProvisioningContextProvider provisioningContextProvider,
-    IUserSecretsManager userSecretsManager)
+    IUserSecretsManager userSecretsManager,
+    IBicepProvisioner bicepProvisioner,
+    IPublishingActivityReporter activityReporter)
 {
-    public async Task DeployModelAsync(CancellationToken cancellationToken = default)
+    public async Task DeployModelAsync(AzureEnvironmentResource resource, CancellationToken cancellationToken = default)
     {
         var userSecrets = await userSecretsManager.LoadUserSecretsAsync(cancellationToken).ConfigureAwait(false);
-        await provisioningContextProvider.CreateProvisioningContextAsync(userSecrets, cancellationToken).ConfigureAwait(false);
+        var provisioningContext = await provisioningContextProvider.CreateProvisioningContextAsync(userSecrets, cancellationToken).ConfigureAwait(false);
+
+        if (resource.PublishingContext is null)
```

Comment:
How does this happen?

Reply:
Currently -- never. I left this as an explicit check for when we do https://github.com/dotnet/aspire/issues/10620.